src/functions/sendCandidateResults/framework/adapter/config/config-adapter.ts

```diff
@@ -42,6 +45,31 @@ export class ConfigAdapter implements IConfigAdapter {
     this.welshLetterFailTemplateId = this.getFromEnvThrowIfNotPresent('NOTIFY_POST_WELSH_FAIL_TEMPLATE_ID');
   }
 
+  public async getApiKey(): Promise<string> {
+    if (!isEmpty(this.apiKey)) {
+      return Promise.resolve(this.apiKey);
+    }
+
+    if (this.isLocal) {
+      return Promise.resolve(this.getFromEnvThrowIfNotPresent('NOTIFY_API_KEY'));
+    }
+
+    const documentServiceSecretName = this.getFromEnvThrowIfNotPresent('DOCUMENTS_SERVICE_SECRET_NAME');
+
+    const secretsmanager = new awsSdk.SecretsManager();
+    const params: awsSdk.SecretsManager.GetSecretValueRequest = {
+      SecretId: documentServiceSecretName,
+    };
+
+    try {
+      const secretValue = await secretsmanager.getSecretValue(params).promise();
+      const secrets = JSON.parse(secretValue.SecretString || '');
```

Comment:
If the default empty string is used here, you will immediately get a NPE on the next line.

Reply:
And the next line after will catch it and reject the procedure. Wouldn't that be a desired outcome?